File: problems/349. Intersection of Two Arrays/solution.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        lookup = {num: True for num in nums1}
        result = []
        for num in nums2:
            if num in lookup and lookup[num]:
                result.append(num)
                lookup[num] = False
        return result

File: problems/349. Intersection of Two Arrays/solution.cpp
class Solution
{
public:
    std::vector<int> intersection(std::vector<int> &num1, std::vector<int> &num2)
    {
        std::vector<int> result;
        std::unordered_map<int, int> lookup;
        for (int num : num1)
            lookup.insert({num, 1});

        for (int num : num2)
        {
            if (lookup[num] == 0)
                continue;
            lookup[num]--;
            result.push_back(num);
        }

        return result;
    }
};